core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -4196,6 +4196,41 @@ object Stream extends StreamLowPriority {
     }
   }
 
+  /** Provides syntax for list of streams. */
+  implicit final class ListStreamOps[F[_], O](private val xs: List[Stream[F, O]]) extends AnyVal {
+
+    /** Nondeterministically merges a (static) list of streams in to a single output stream.
+      *
+      * When any of the merged streams fail, then the output stream and all other inner
+      * streams are interrupted, resulting in a stream that fails with the error of the
+      * stream that caused initial failure.
+      *
+      * Finalizers on each stream are run at the end of the stream,
+      * concurrently with other stream computations.
+      *
+      * Finalizers on the output stream are run after the output stream has finished
+      * (i.e., all open inner streams have finished).
+      *
+      * See [[NestedStreamOps.parJoinUnbounded]] for a strictly more powerful (albeit slower) variant
+      * capable of merging a stream of streams.
+      */
+    def parJoinUnbounded(implicit F: Concurrent[F]): Stream[F, O] =
+      if (xs.size <= 1) xs.headOption.getOrElse(Stream.empty)
+      else {
+        Stream.eval((Channel.bounded[F, Chunk[O]](64), F.deferred[Unit]).tupled).flatMap {
```

Comment:
Is this 64 arbitrary?

Reply:
It is. `Stream#parJoin` uses `Channel.synchronous`, `Channel.bounded` gives s speed bump, and I could not find a test procedure which distincts the two.

Reply:
I tried the snippet and it never fails. With the removed `.timeoutTo` the results are as follows:

```scala
      IO.ref(0).flatMap { ref =>
        Stream(
          Stream.repeatEval(ref.getAndUpdate(_ + 1).void),
          Stream.empty,
        ).parJoinUnbounded.head.compile.drain >> ref.get.assertEquals(1)
      }
```
fails with `1 != 2`,
```scala
      IO.ref(0).flatMap { ref =>
        List(
          Stream.repeatEval(ref.getAndUpdate(_ + 1).void),
          Stream.empty,
        ).parJoinUnbounded.head.compile.drain >> ref.get.assertEquals(1)
      }
```
fails with `1 != 5`.

Reply:
> I tried the snippet and it never fails. With the removed `.timeoutTo`

Oh sorry, you're right, the `timeoutTo` doesn't make sense :)

In any case, it seems like it should be `assertEquals(2)` then. Does your `List` version _always_ complete with `5`? Seems like a race condition so I expect you to get inconsistent values.

Reply:
Yes, 5 is arbitrary. If  it must return 2, we shall revert to Channel.synchronous. And add proper tests :)

Reply:
@armanbilge I switched to `Channel.synchronous` and added the tests. Still, the results are flaky - `ref.get` returns `1` or `2` (even for `Stream#parJoinUnbounded`). Both tests assert `ref.get <= 2`.

Reply:
@armanbilge what do you think in general?